### PR TITLE
Clean up RecordDispatcher

### PR DESCRIPTION
**Pull request by pierDipi**
Nits I noticed reviewing https://github.com/knative-sandbox/eventing-kafka-broker/pull/614.

## Proposed Changes

- Remove unused constructor
- Regenerate the third party file

### CODE REVIEW

**Comment by pierDipi** [comment]
[Sacura failed, I'll take a look](https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/620/pull-knative-sandbox-eventing-kafka-broker-integration-tests/1359265311584423936)
/test pull-knative-sandbox-eventing-kafka-broker-integration-tests

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java [hunk 1]
```diff
@@ -184,8 +162,10 @@ private void onSubscriberSuccess(final KafkaConsumerRecord<String, CloudEvent> r
       .onComplete(finalProm);
   }
 
-  private void onSubscriberFailure(final KafkaConsumerRecord<String, CloudEvent> record, final Throwable exception,
+  private void onSubscriberFailure(final KafkaConsumerRecord<String, CloudEvent> record,
+                                   final Throwable exception,
                                    final Promise<Void> finalProm) {
+    logError("Failed to send event to subscriber", record, exception);
```
Comment (slinkydeveloper) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java:168]: No need for this log line, it's already in `composeSenderAndSinkHandler`

Comment (pierDipi) [reply data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java]: Done!

**Comment by slinkydeveloper** [comment]
/lgtm
/approve

Thanks for the fixes

**Pull request merged**